test/utils/asserts.hpp

```diff
@@ -1,7 +1,23 @@
 #pragma once
 
+#include <gmock/gmock.h>
 #include <gtest/gtest.h>
 
+MATCHER(Traps, "")
+{
+    return arg.trapped == true;
+}
+
+MATCHER(Result, "empty result")
+{
+    return arg.trapped == false && arg.stack.size() == 0;
+}
+
+MATCHER_P(Result, value, "single result")
```

Comment:
```suggestion
MATCHER_P(Result, value, "")
```
Leave the description empty so the it will display e.g. `result 2` for `Result(2)`.